assembly.py

```diff
@@ -339,6 +340,55 @@ def parser_assemble_trinity(parser=argparse.ArgumentParser()):
 __commands__.append(('assemble_trinity', parser_assemble_trinity))
 
 
+def assemble_spades(
+    inBam,
+    outFasta,
+    spades_opts='',
+    previously_assembled_contigs=None,
+    mem_limit_gb=4,
+    threads=1,
+):
+    ''' De novo RNA-seq assembly with the SPAdes assembler.
+
+    Inputs:
+        inBam - reads to assemble.  May include both paired and unpaired reads.
+        previously_assembled_contigs - (optional) already-assembled contigs from the same sample.
+
+    Outputs:
+        outFasta - the assembled contigs.  Note that, since this is RNA-seq assembly, for each assembled genomic region there may be
+            several contigs representing different variants of that region.
+
+    Params:
+        mem_limit_gb - max memory to use
+        threads - number of threads to use
+        spades_opts - (advanced) custom command-line options to pass to the SPAdes assembler
+
+    '''
+
+    with tools.picard.SamToFastqTool().execute_tmp(inBam, includeUnpaired=True) as (reads_fwd, reads_bwd, reads_unpaired):
```

Comment:
Similar questions to other PR:
- if unspecified or set to None, can Spades do something reasonable on its own for thread count and memory limit?

Also, if we have a memory limit suggestion given to us already, perhaps we should be passing that to the JVMmemory optional param when running Picard?